src/collection-preferences/content-display/index.tsx

```diff
@@ -52,20 +52,30 @@ export default function ContentDisplayPreference({
   const i18n = useInternalI18n('collection-preferences');
   const [columnFilteringText, setColumnFilteringText] = useState('');
 
-  const onToggle = (option: OptionWithVisibility) => {
-    onChange(value.map(item => (item.id === option.id ? { ...item, visible: !option.visible } : item)));
-  };
-
   const titleId = `${idPrefix}-title`;
   const descriptionId = `${idPrefix}-description`;
 
-  const sortedAndFilteredOptions = useMemo(
-    () =>
-      getSortedOptions({ options, contentDisplay: value }).filter(option =>
-        option.label.toLowerCase().trim().includes(columnFilteringText.toLowerCase().trim())
-      ),
-    [columnFilteringText, options, value]
-  );
+  const [sortedOptions, sortedAndFilteredOptions] = useMemo(() => {
+    const sorted = getSortedOptions({ options, contentDisplay: value });
+    const filtered = sorted.filter(option =>
+      option.label.toLowerCase().trim().includes(columnFilteringText.toLowerCase().trim())
+    );
+    return [sorted, filtered];
+  }, [columnFilteringText, options, value]);
+
+  const addMissingOptionsInValue = () => {
```

Comment:
I'm not sure about this pattern of having to call this `addMissingOptionsInValue` function in multiple places before calling `onChange`, could we instead use `sortedOptions` directly in place of `value` throughout this component? Or create a function that wraps `onChange` and does the addMissing in just one place?

Reply:
Good call, I was deep in the weeds when I wrote this and reading it again, it does look weird. I'll fix that and submit a new revision right now.